scipy/optimize/_qap.py

```diff
@@ -507,6 +519,29 @@ def _quadratic_assignment_faq(A, B,
     return OptimizeResult(res)
 
 
+def _adj_pad(A, B, method):
+    # pads the matrix with less nodes such that A & B are same size
+    # schemes according to section 2.5 of [2]
```

Comment:
Yes, this does seem to do the following:
![image](https://user-images.githubusercontent.com/6570539/98703618-8c0d0d00-2330-11eb-955c-3ab435ad12c6.png)